repository/src/test/java/org/alfresco/schedule/Cleaner.java

```diff
@@ -0,0 +1,179 @@
+            // cleaner is not lost in case of any problems encountered during the job execution
+            AuthenticationUtil.runAsSystem(() -> {
+                RetryingTransactionCallback<Void> txnWork = () -> {
+                    try
+                    {
+                        nodeService.deleteNode(nodeRef);
+                    }
+                    catch (InvalidNodeRefException inre)
+                    {
+                        handleError(inre);
+                    }
+                    deletedNodes.getAndIncrement();
+                    // Waiting REMOVAL_WAIT_TIME_MS seconds for next deletion so we don't need to have many nodes on the trash can
+                    Thread.sleep(REMOVAL_WAIT_TIME_MS);
+                    return null;
+                };
+                return transactionService.getRetryingTransactionHelper().doInTransaction(txnWork, false, true);
+            });
+        }
+        return deletedNodes.get();
+    }
+
+    /**
+     *
+     * It returns the {@link java.util.List List} of {@link org.alfresco.service.cmr.repository.NodeRef NodeRef} of the
+     * archive store set to be deleted according to configuration for <b>deleteBatchCount</b>.
+     *
+     * @return
+     */
+    private List<NodeRef> getBatchToDelete()
+    {
+        return getChildAssocs().stream().map(ChildAssociationRef::getChildRef).collect(Collectors.toList());
+    }
+
+    /**
+     *
+     * It will return the first {@link #deleteBatchCount}
+     * {@link org.alfresco.service.cmr.repository.ChildAssociationRef}s of type {@link ContentModel}.ASSOC_CHILDREN from
+     * the archive store set.
+     *
+     * @return
+     */
+    private List<ChildAssociationRef> getChildAssocs()
+    {
+        StoreRef archiveStore = new StoreRef(archiveStoreUrl);
+        NodeRef archiveRoot = nodeService.getRootNode(archiveStore);
+        return nodeService.getChildAssocs(archiveRoot, ContentModel.ASSOC_CHILDREN, RegexQNamePattern.MATCH_ALL, deleteBatchCount,
+                false);
+    }
+
+    /**
+     *
+     * The method that will clean the specified <b>archiveStoreUrl</b> to the limits defined by the values set for
+     * <b>deleteBatchCount</b>.
+     */
+    public void clean()
+    {
+        LOGGER.info("Running TestCleaner");
+
+        // Retrieve in a new read-only transaction the list of nodes to be deleted by the TestCleaner
+        AuthenticationUtil.runAsSystem(() -> {
+            RetryingTransactionCallback<Void> txnWork = () -> {
+                nodesToClean = getBatchToDelete();
+
+                LOGGER.info(String.format("Number of nodes to delete: %s", nodesToClean.size()));
+
+                return null;
+            };
+            return transactionService.getRetryingTransactionHelper().doInTransaction(txnWork, true, true);
+        });
+
+        int deletedNodes = deleteNodes(nodesToClean);
+
+        LOGGER.info(String.format("TestCleaner finished. Number of deleted nodes: %s", deletedNodes));
```

Comment:
Against this would be simplified with slf4j:
```
LOGGER.info("TestCleaner finished ... nodes {}", deletedNodes);
```

Reply:
Simplified `log`statement as suggested.